### ISSUE TITLE
Fluid Drawer took too long to load model (~7 seconds)

### ISSUE THREAD

**Issue by Krutoy242** [phantamanta44/fluid-drawers#19]
I profiled my game load `PreInit` stage. I found that Fluid Drawers, despite having only one block consume too much time to load. Its about ~7 seconds and more than most of other ~400 mods i have in my modpack.

![image](https://github.com/phantamanta44/fluid-drawers/assets/936618/8771082c-4126-4f96-a736-a94b53209f63)

## https://spark.lucko.me/ZW1KynfgAK

Is it normal behavior? How can i improve that and make it load faster?

**Comment by phantamanta44** [comment]
I guess the problem is that my model loader loads every model file in the game to check if it should handle loading that model; I assumed that there wouldn't ever be enough models for that to be a significant performance issue, but I guess e2e:e is big enough that it becomes one. I'm occupied for a bit, but I might be able to get a patch out in a couple weeks

**Comment by Krutoy242** [comment]
Would be great! I'm happy to wait.

Also in the profile you can see several other methods that have heavy loads to. They all related to `libnine` model calls.

It's it would be helpful if I pick them from profile and write down as separate list?

**Comment by phantamanta44** [comment]
nah, those are all exactly the same issue; fluid drawers loads models exactly the same way l9 does

**Comment by Krutoy242** [comment]
> but I might be able to get a patch out in a couple weeks

Still waiting! 😃
